internal/databases/postgres/extract_provider_with_spec.go

```diff
@@ -0,0 +1,90 @@
+package postgres
+
+import (
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/wal-g/wal-g/internal"
+)
+
+const (
+	defaultTbspPrefix = "/" + DefaultTablespace + "/"
+	customTbspPrefix  = "/" + TablespaceFolder + "/"
+	systemIDLimit     = 16384
+)
+
+type ExtractProviderDBSpec struct {
+	ExtractProviderImpl
+	onlyDatabases []string
+}
+
+func NewExtractProviderDBSpec(onlyDatabases []string) *ExtractProviderDBSpec {
+	return &ExtractProviderDBSpec{onlyDatabases: onlyDatabases}
+}
+
+func (p ExtractProviderDBSpec) Get(
+	backup Backup,
+	filesToUnwrap map[string]bool,
+	skipRedundantTars bool,
+	dbDataDir string,
+	createNewIncrementalFiles bool,
+) (IncrementalTarInterpreter, []internal.ReaderMaker, string, error) {
+	_, filesMeta, _ := backup.GetSentinelAndFilesMetadata()
+
+	fullRestoreDatabases := p.makeFullRestoreDatabaseMap(p.onlyDatabases, filesMeta.DatabasesByNames)
+	p.filterFilesToUnwrap(filesToUnwrap, fullRestoreDatabases)
+
+	return p.ExtractProviderImpl.Get(backup, filesToUnwrap, skipRedundantTars, dbDataDir, createNewIncrementalFiles)
+}
+
+func (p ExtractProviderDBSpec) makeFullRestoreDatabaseMap(databases []string, names DatabasesByNames) map[int]bool {
+	restoredDatabases := p.makeSystemDatabasesMap()
+
+	for _, db := range databases {
+		dbID, err := names.Resolve(db)
+		if err == nil {
```

Comment:
What if we failed to resolve the ID? I guess that we should not ignore it silently.